### PR TITLE
[mdns-mdnssd] support mDNS service removing

### PR DESCRIPTION
**Pull request by simonlingoogle**
Based on #1122

### CODE REVIEW

File: src/utils/dns_utils.cpp [hunk 1]
```diff
@@ -69,6 +70,28 @@ std::string UnescapeInstanceName(const std::string &aName)
     return newName;
 }
 
+void CheckHostnameSanity(const std::string &aHostName)
+{
+    OTBR_UNUSED_VARIABLE(aHostName);
+
+    assert(aHostName.length() > 0);
+    assert(aHostName[aHostName.length() - 1] == '.');
```
Comment (superwhd) [inline src/utils/dns_utils.cpp:78]: nit: use `aHostName.back()`

File: src/utils/dns_utils.cpp [hunk 2]
```diff
@@ -69,6 +70,28 @@ std::string UnescapeInstanceName(const std::string &aName)
...
+
+void CheckServiceNameSanity(const std::string &aType)
+{
+    size_t dotpos;
+
+    OTBR_UNUSED_VARIABLE(aType);
+    OTBR_UNUSED_VARIABLE(dotpos);
+
+    assert(aType.length() > 0);
+    assert(aType[aType.length() - 1] != '.');
```
Comment (superwhd) [inline src/utils/dns_utils.cpp:89]: nit: use `aHostName.back()`

File: src/utils/dns_utils.hpp [hunk 3]
```diff
@@ -55,6 +55,29 @@ namespace DnsUtils {
...
+ * @param[in] aHostName  The host name to check.
+ *
+ */
+void CheckHostnameSanity(const std::string &aHostName);
+
+/**
+ * This function checks a given service name for sanity.
+ *
+ * Check criteria:
+ *      The service name must contain one dot.
```
Comment (superwhd) [inline src/utils/dns_utils.hpp:73]: `The service name must contain exactly one dot.`

File: src/utils/dns_utils.hpp [hunk 4]
```diff
@@ -55,6 +55,29 @@ namespace DnsUtils {
...
+ *      The service name must not end with dot.
+ *
+ * @param[in] aType  The service name to check.
+ *
+ */
+void CheckServiceNameSanity(const std::string &aType);
```
Comment (superwhd) [inline src/utils/dns_utils.hpp:79]: Shall we name it `CheckServiceTypeSanity`?

File: src/utils/dns_utils.cpp (hunk 1 above)
Comment (superwhd) [reply src/utils/dns_utils.cpp:78]: Just realized that these methods aren't introduced in this PR. Feel free to leave them unchanged.

Comment (simonlingoogle) [reply src/utils/dns_utils.cpp]: fixed to use .back()

File: src/utils/dns_utils.hpp (hunk 4 above)
Comment (simonlingoogle) [reply src/utils/dns_utils.hpp]: Changed `aType` to `aServiceName`.

**Review by superwhd — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**